### PR TITLE
Remove bcypto

### LINKED ISSUE #93: Replace bcrypto dep
We already tried this once and it didn't work but it was some time ago and some alternatives appeared.

Our bcrypto algo dependencies and proposed alternatives:
- x25519 
   - https://www.npmjs.com/package/ecies-25519
- aead (chacha-poly)
   -  https://www.npmjs.com/package/@stablelib/chacha20poly1305 (https://github.com/StableLib/stablelib/blob/master/packages/xchacha20poly1305/xchacha20poly1305.ts)
   - https://www.npmjs.com/package/chacha-js
- hkdf
   - https://www.npmjs.com/package/futoin-hkdf
   - https://www.npmjs.com/package/js-crypto-hkdf
   - https://www.npmjs.com/package/@stablelib/hkdf

### PR DESCRIPTION
**Pull request by mpetrun5**
Replaced:
 - bcrypto/x25519 with  https://github.com/StableLib/stablelib/tree/master/packages/x25519
 - bcrypto/aead with https://github.com/StableLib/stablelib/tree/master/packages/chacha20poly1305
 - bcrypto/hkdf with https://github.com/StableLib/stablelib/tree/master/packages/hkdf

resolves #93

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -36,6 +36,7 @@
     "util": false
   },
   "devDependencies": {
+    "@types/elliptic": "^6.4.12",
```
Comment (mpetrunic) [inline package.json:39]: I don't think you really need this.

File: src/handshakes/abstract-handshake.ts [hunk 2]
```diff
@@ -80,15 +75,27 @@ export abstract class AbstractHandshake {
...
-    ciphertext = ciphertext.slice(0, ciphertext.length - 16)
-    ctx.init(k, nonce)
-    ctx.aad(ad)
-    ctx.decrypt(ciphertext)
-    // Decryption is done on the sent reference
-    return { plaintext: ciphertext, valid: ctx.verify(tag) }
+    const ctx = new ChaCha20Poly1305(k)
+    const encryptedMessage = ctx.open(
+      nonce,
+      new Uint8Array(ciphertext.buffer, ciphertext.byteOffset, ciphertext.length),
```
Comment (mpetrunic) [inline src/handshakes/abstract-handshake.ts:81]: buffer is subclass of Uint8Array so you shouldn't be required to create new one

File: src/noise.ts [hunk 3]
```diff
@@ -49,10 +49,18 @@ export class Noise implements INoiseConnection {
     this.useNoisePipes = false
 
     if (staticNoiseKey) {
-      const publicKey = x25519.publicKeyCreate(staticNoiseKey)
+      const keyPair = x25519.generateKeyPairFromSeed(staticNoiseKey)
```
Comment (mpetrunic) [inline src/noise.ts:52]: this seems wrong, given key is privateKey not seed imo

Comment (mpetrun5) [reply src/noise.ts:52]: Its called generateKeyPairFromSeed but it actually accepts SECRET_KEY_LENGTH of 32 which is not seed but actually the 'staticNoiseKey'. 
Maybe I misread something?

Comment (mpetrunic) [reply src/noise.ts:52]: you are right, weird naming. Just add code comment as to not confuse other people

**Review by mpetrunic — APPROVED** [review]

**Comment by vasco-santos** [comment]
🎉 

Do you have any metrics on the bundle size difference?

**Comment by vasco-santos** [comment]
~27% decrease on its bundle size, from 191.5kb to 141.2kb (minified+gzipped)

🚀 ❤️

**Comment by mpetrunic** [comment]
> ~27% decrease on its bundle size, from 191.5kb to 141.2kb (minified+gzipped)
> 
> 🚀 ❤️

from our benchmark there is also speed improvment:
```
NEW:

Initializing handshake benchmark
Init complete, running benchmark
handshake x 39.97 ops/sec ±4.98% (67 runs sampled)
Done in 6.04s.


OLD:

Initializing handshake benchmark
Init complete, running benchmark
handshake x 27.23 ops/sec ±4.57% (67 runs sampled)
Done in 6.10s.
```

**Pull request merged**